examples/auction/src/lib.rs

```diff
@@ -125,7 +125,7 @@ fn auction_bid<S: HasStateApi>(
     };
 
     // Ensure that the new bid exceeds the highest bid so far
-    ensure!(amount > balance, BidError::BidTooLow);
+    ensure!(amount > balance - amount, BidError::BidTooLow);
```

Comment:
I can see this is rigth, but it does look puzzling :)

Reply:
Maybe introduce a new variable
```
// Balance of the contract before the call
let previous_balance = balance - amount;
```

Reply:
addressed